### PR TITLE
MapCanvas cleanup + resurrection of mouse wheel and touble tap zoom in

### PR DESCRIPTION
**Pull request by nirvn**
Fixes and cleanup

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 3e099b0cbe58385655f4037a10431ef0226a63b9:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-f38861de9205a771553ae03da4e5c33482636547-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-f38861de9205a771553ae03da4e5c33482636547-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-f38861de9205a771553ae03da4e5c33482636547-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-f38861de9205a771553ae03da4e5c33482636547-x86.apk)

**Comment by nirvn** [comment]
Just tested, it's sweet to get our double tap to zoom in again.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 455db6b4b8a96893b64f1efa17965fa744b54e6a:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-425ed41a3c4719d33f3e5fb6f85815935c2ff6e0-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-425ed41a3c4719d33f3e5fb6f85815935c2ff6e0-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-425ed41a3c4719d33f3e5fb6f85815935c2ff6e0-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-425ed41a3c4719d33f3e5fb6f85815935c2ff6e0-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 7bcda5885b02b778cd1e1d86d48a247336287543:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-0628af4767ab190e100416bc2d5b1e4eef3648e6-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-0628af4767ab190e100416bc2d5b1e4eef3648e6-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-0628af4767ab190e100416bc2d5b1e4eef3648e6-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-0628af4767ab190e100416bc2d5b1e4eef3648e6-x86.apk)

File: src/qml/MapCanvas.qml [hunk 1]
```diff
@@ -264,5 +285,20 @@ Item {
         }
     }
 
+
+    WheelHandler {
```
Comment (m-kuhn) [inline src/qml/MapCanvas.qml:289]: Finally!

**Comment by nyalldawson** [comment]
This is very smoothly, thanks! One minor thing is that the direction is reversed compared to Google maps (in which up =zoom out). I think it's important to follow the Google maps convention here, just for users familiarity/muscle memory...

**Comment by nirvn** [comment]
@nyalldawson , fair point, zoom in<->out reversed. Thanks for testing.

**Comment by nirvn** [comment]
@nyalldawson , fair point, zoom in<->out reversed. Thanks for testing.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 8b474b44201c2119279e8826fcde9e0bd5171fb8:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-b49d6d4b546a3815f98e7e026cc7432710088da9-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-b49d6d4b546a3815f98e7e026cc7432710088da9-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-b49d6d4b546a3815f98e7e026cc7432710088da9-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1537-b49d6d4b546a3815f98e7e026cc7432710088da9-x86.apk)

File: src/qml/qgismobileapp.qml [hunk 2]
```diff
@@ -279,27 +279,44 @@ ApplicationWindow {
...
-          // for instance, the vertex editor will select a vertex if possible
-        }
-        else if ( type === "stylus" && ( ( stateMachine.state === "digitize" && dashBoard.currentLayer ) || stateMachine.state === 'measure' ) ) {
-          if ( Number( currentRubberband.model.geometryType ) === QgsWkbTypes.PointGeometry ||
-              Number( currentRubberband.model.geometryType ) === QgsWkbTypes.NullGeometry )
+          // Check if geometry editor is taking over
+          if ( geometryEditorsToolbar.canvasClicked(point) )
+              return;
+
+          if (locatorItem.state == "on")
```
Comment (m-kuhn) [inline src/qml/qgismobileapp.qml:286]: Is this correct, the locator should stay open if the geometry editor is taking over?

Comment (nirvn) [reply src/qml/qgismobileapp.qml:286]: No, but the locator shouldn't be open if the geometry editor is on (which is a statement that isn't valid ATM, we can fix in a different PR).

**Review by suricactus — APPROVED** [review]

**Pull request merged**